preprocessors/semanticSeg/segment.py

```diff
@@ -137,10 +137,31 @@ def run_segmentation(url,
     pil_image = cv2.imdecode(image, cv2.IMREAD_COLOR)
     height, width, channels = pil_image.shape
     if(height > 1500 or width > 1500):
-        pil_image = cv2.resize(pil_image, (1500, 1500),
-                               interpolation=cv2.INTER_AREA)
-        width = 1500
-        height = 1500
+        if(height > 1500 and width < 1500):
+            scale_size = np.float(1500.0 / np.float(height))
+            height = np.int(height * scale_size)
+            width = np.int(width * scale_size)
+            pil_image = cv2.resize(pil_image, (width, height),
+                                   interpolation=cv2.INTER_AREA)
+        elif(width > 1500 and height < 1500):
+            scale_size = np.float(1500.0 / np.float(width))
+            height = np.int(height * scale_size)
+            width = np.int(width * scale_size)
+            pil_image = cv2.resize(pil_image, (width, height),
+                                   interpolation=cv2.INTER_AREA)
+        else:
+            if(height > width):
+                scale_size = np.float(1500.0 / np.float(height))
+                height = np.int(height * scale_size)
+                width = np.int(width * scale_size)
+                pil_image = cv2.resize(pil_image, (width, height),
+                                       interpolation=cv2.INTER_AREA)
+            else:
+                scale_size = np.float(1500.0 / np.float(width))
+                height = np.int(height * scale_size)
+                width = np.int(width * scale_size)
+                pil_image = cv2.resize(pil_image, (width, height),
+                                       interpolation=cv2.INTER_AREA)
```

Comment:
This is a large amount of code for the task, with a constant (1500) hardcoded in many different places, with multiple nested if/elses that are easy to get wrong and hard to read due to repetitive code. I think something like the following (uncompiled! not tested!) would be both clearer and much shorter, so less prone to bugs if it needs to be changed in the future:

```
  scale_size = np.float(1500 / np.float(max(height,width))
     if(scale_size < 1.0):    # graphic too big! it needs to be scaled down
        height = np.int(height * scale_size)
        width = np.int(width * scale_size)
        pil_image = cv2.resize(pil_image, (width, height),
                            interpolation=cv2.INTER_AREA)
       #safer and probably faster to just get new dimensions
       #  from the actual new image, e.g., avoids rounding differences
       height, width, channels = pil_image.shape
```